### PR TITLE
opt in tracking DRAFT

### PR DESCRIPTION
**Pull request by nickreynolds**
- use cookie to track
- add middleware dynamically when cookie found/set

### CODE REVIEW

File: packages/dapp/src/components/Main.tsx [hunk 1]
```diff
@@ -39,22 +39,38 @@ import WrongNetwork from "./WrongNetwork";
...
+import { addMiddleware } from "redux-dynamic-middlewares";
+import { gaMiddleware } from "../redux/analytics";
 
 export interface MainReduxProps {
   network: string;
 }
 
 export interface MainState {
   prevAccount: EthAddress;
+  cookieSet: boolean;
```
Comment (am9u) [inline packages/dapp/src/components/Main.tsx]: Is this actually being used?

File: packages/dapp/src/components/Main.tsx [hunk 2]
```diff
@@ -118,10 +134,25 @@ class Main extends React.Component<MainReduxProps & DispatchProp<any> & RouteCom
...
+  public allowTracking = (ev: any): void => {
+    this.answerTracking();
+    cookie.save("allowTracking", true, { path: "/" });
+    addMiddleware(gaMiddleware);
+    this.setState({ cookieSet: true });
+  };
+
+  public denyTracking = (ev: any): void => {
+    this.answerTracking();
+    cookie.save("allowTracking", false, { path: "/" });
```
Comment (am9u) [inline packages/dapp/src/components/Main.tsx]: I think we might want the `allowTracking` cookie to use the same `maxAge` value as the `didAnswerTracking` cookie. And maybe set that as a variable or constant in this file.

File: packages/dapp/src/components/Main.tsx (hunk 1 above)
Comment (nickreynolds) [reply packages/dapp/src/components/Main.tsx]: kind of. it just modifies the state so we do a re-render and get rid of the AskTrackingBar

Comment (nickreynolds) [reply packages/dapp/src/components/Main.tsx]: but the value of it isn't used

Comment (nickreynolds) [reply packages/dapp/src/components/Main.tsx]: i guess `.forceUpdate()` is better, right?

File: packages/dapp/src/components/Main.tsx (hunk 2 above)
Comment (nickreynolds) [reply packages/dapp/src/components/Main.tsx]: makes sense

File: packages/dapp/src/redux/store.ts [hunk 3]
```diff
@@ -2,14 +2,31 @@ import thunk from "redux-thunk";
...
+export const store = () => {
+  const didAnswerTracking = cookie.load("didAnswerTracking");
+  if (didAnswerTracking) {
+    const allowTracking = cookie.load("allowTracking");
+    if (allowTracking) {
+      console.log("THIS ONE");
+      return createStore(
+        createRootReducer(history),
+        {},
+        compose(applyMiddleware(routerMiddleware(history), thunk, gaMiddleware)),
```
Comment (walfly) [inline packages/dapp/src/redux/store.ts]: if this middleware isnt there what does trackpageview do?

Comment (nickreynolds) [reply packages/dapp/src/redux/store.ts]: nothing, when the middleware isn't there no actions are tracked. when the middleware is added it picks them up again.

**Pull request closed without merging**